website/source/guides/node-draining.html.md

```diff
@@ -0,0 +1,247 @@
+2018-04-11T23:42:27Z: Alloc "78b6b393-d29c-d8f8-e8e8-28931c0013ee" status running -> complete
+2018-04-11T23:42:29Z: Alloc "9a51a484-8c43-aa4e-d60a-46cfd1450780" marked for migration
+2018-04-11T23:42:29Z: Alloc "9a51a484-8c43-aa4e-d60a-46cfd1450780" draining
+2018-04-11T23:42:29Z: Node "46f1c6c4-a0e5-21f6-fd5c-d76c3d84e806" drain complete
+2018-04-11T23:42:34Z: Alloc "9a51a484-8c43-aa4e-d60a-46cfd1450780" status running -> complete
+2018-04-11T23:42:34Z: All allocations on node "46f1c6c4-a0e5-21f6-fd5c-d76c3d84e806" have stopped.
+```
+
+There are a couple important events to notice in the output. First, only 2
+allocations are migrated initially:
+
+```
+2018-04-11T23:41:57Z: Alloc "5b4d6db5-3fcb-eb7d-0415-23eefcd78b6a" marked for migration
+2018-04-11T23:41:57Z: Alloc "56f770c0-f8aa-4565-086d-01faa977f82d" marked for migration
+```
+
+This is because `max_parallel = 2` in the job specification. The next
+allocation on the draining node waits to be migrated:
+
+```
+2018-04-11T23:42:22Z: Alloc "78b6b393-d29c-d8f8-e8e8-28931c0013ee" marked for migration
+```
+
+Note that this occurs 25 seconds after the initial migrations. The 25 second
+delay is because a replacement allocation took 10 seconds to become healthy and
+then the `min_healthy_deadline = "15s"` meant node draining waited an
+additional 15 seconds. If the replacement allocation had failed within that
+time the node drain would not have continued until a replacement could be
+successfully made.
+
+### Scheduling Eligibility
+
+Now that the example drain has finished we can inspect the state of the drained
+node:
+
+```text
+$ nomad node status
+ID        DC   Name     Class   Drain  Eligibility  Status
+46f1c6c4  dc1  nomad-5  <none>  false  ineligible   ready
+96b52ad8  dc1  nomad-6  <none>  false  eligible     ready
+f7476465  dc1  nomad-4  <none>  false  eligible     ready
+```
+
+While node `46f1` has `Drain = false`, notice that its `Eligibility =
+ineligible`. Node scheduling eligibility is a new field in Nomad 0.8. When a
+node is ineligible for scheduling the scheduler will not consider it for new
+placements.
+
+While draining, a node will always be ineligible for scheduling. Once draining
+completes it will remain ineligible to prevent refilling a newly drained node.
+
+However, by default canceling a drain with the `-disable` option will reset a
+node to be eligible for scheduling. To cancel a drain and preserving the node's
+ineligible status use the `-keep-ineligible` option.
+
+Scheduling eligibility can be toggled independently of node drains by using the
+[`nomad node eligibility`][eligibility] command.
+
+### Node Drain Deadline
+
+Sometimes a drain is unable to proceed and complete normally. This could be
+caused by not enough capacity existing in the cluster to replace the drained
+allocations or by replacement allocations failing to start successfully in a
+timely fashion.
+
+Operators may specify a deadline using the option for node drain to prevent
```

Comment:
Operators may specify a deadline when enabling a node drain to prevent drains from not finishing.